app/assets/javascripts/exercises.coffee

```diff
@@ -418,7 +424,142 @@ ready =->
     $('.primary-checkbox').prop('checked', false)
     $(event.target).prop('checked', true)
 
-  if $('.primary-checkbox:checked').length < 1
+  $('body').on 'click', '.export-action', (event) ->
+    exportType = $(this).attr('data-export-type')
+    exerciseID = $(this).parents('.import-export-exercise').attr('data-exercise-id')
+    accountLinkID = $(this).parents('.import-export-exercise').attr('data-account-link')
+    exportExerciseConfirm(exerciseID, accountLinkID, exportType)
+
+  $('body').on 'click', '.export-close-button', (event) ->
+    $(this).parents('.import-export-exercise').remove()
+
+  $('body').on 'click', '.import-action', (event) ->
+    importId = $(this).parents('.import-export-exercise').attr('data-import-id')
+    subfileId = $(this).parents('.import-export-exercise').attr('data-import-subfile-id')
+    importType = $(this).attr('data-import-type')
+    importExerciseConfirm(importId, subfileId, importType)
+
+  $('#exercise_execution_environment_id').on 'change', ->
+    checkExecutionEnvironment()
+
+  $('input:radio[name="exercise[private]"]').on 'change', ->
+    checkExecutionEnvironment()
+
+  if $('.primary-checkbox').length > 0 && $('.primary-checkbox:checked').length < 1
     $('.primary-checkbox')[0].click()
 
 $(document).on('turbolinks:load', ready)
+
+checkExecutionEnvironment = () ->
+  executionEnvironment = $('#exercise_execution_environment_id').val()
+  visibilityPrivate = $('input:radio:checked[name="exercise[private]"]').val()
+
+  if executionEnvironment == '' && visibilityPrivate == 'false'
+    console.log 'show'
+    $('.execution-environment-error').removeClass('hidden')
+    $('.execution-environment-error').addClass('errors')
+    $('.execution-environment-error').parent().find('.form-label').addClass('errors')
+  else
+    console.log 'hide'
+    $('.execution-environment-error').addClass('hidden')
+    $('.execution-environment-error').removeClass('errors')
+    $('.execution-environment-error').parent().find('.form-label').removeClass('errors')
+
+importExerciseStart = () -> #maybe unobstrusive like export?
+  # console.log('start')
+  formData = new FormData()
+  formData.append('zip_file', document.getElementById('exercise_import').files[0])
+
+  $.ajax({
+    type: 'POST',
+    url: '/exercises/import_exercise_start',
+    data: formData,
+    processData: false,
+    contentType: false,
+    # success: (response) ->
+      # console.log(response)
```

Comment:
Is here something missing? This request is used to stat the import process while the other one below is used to grab the import status, right?

Reply:
This one starts the dialog while `import_exercise_confirm` starts the actual import process.

Reply:
Alright, I see. Thanks for clarification :)